File: e2e/sdk-matrix/dart/bin/main.dart
import 'dart:io';

import 'package:sentry/sentry.dart';

class a {
  static Future<void> b() async {
    await Sentry.captureException(
      StateError('Real Dart SDK compatibility probe'),
      stackTrace: StackTrace.current,
    );
  }
}

Future<void> main() async {
  await Sentry.init((options) {
    options.dsn = Platform.environment['SPICYTRACK_DSN']!;
    options.environment = 'sdk-matrix';
    options.release = 'sdk-dart@9.26.0';
  });
  await a.b();
  await Sentry.close();
}

File: e2e/sdk-matrix/rust/src/main.rs
use std::{borrow::Cow, io, time::Duration};

fn main() {
    let dsn = std::env::var("SPICYTRACK_DSN").expect("SPICYTRACK_DSN is required");
    let mut options = sentry::ClientOptions::default();
    options.environment = Some(Cow::Borrowed("sdk-matrix"));
    options.release = Some(Cow::Borrowed("sdk-rust@0.49.1"));
    let guard = sentry::init((dsn, options));
    sentry::capture_error(&io::Error::other("Real Rust SDK compatibility probe"));
    guard.close(Some(Duration::from_secs(10)));
}
